src/euphorie/client/docx/compiler.py

```diff
@@ -732,13 +779,82 @@ def compile(self, data):
         Check the file .../daimler/oira/tests/mocked_data.py
         to understand its format
         '''
+        self.unanwered_risks = []
+        self.not_applicable_risks = []
```

Comment:
Can `unanwered_risks` and `not_applicable_risks` be properties?

Reply:
What would be the benefit? Those lists get filled in `set_modules_rows`. Do you think a property with a setter would be better?

Reply:
The benefit is code clarity.
Why those attributes are defined here if they are filled in `set_modules_rows`?
I might be missing something trivial but code like this smells to me.

Reply:
> So do you think that writing a property method with a setter would be appropriate here? 

No

> Or do you have another suggestion?

I do not know, it seems to me that the `set_modules_rows` is overly complex.
I am wondering why, for example, the data structure you are passing does not already have this partition.
But anyway, I do not want to block your work.

Reply:
The data structure is created here: https://github.com/euphorie/Euphorie/blob/fb16d0be4eb3b9c5d6d41f30fc1ef685793d1944/src/euphorie/client/docx/views.py#L84

That means, for every module, it lists all risks in their given order. Of course, I could already create the partition in the method `get_risks_for`, so that the return value might be a tuple (risks, unanswered_risks, not_applicable_risks). 
But then, `set_modules_rows` would become even more complex. If `only_anwered_risks` is False, then I would need to iterate both over risks and unanswered_risks, and make sure that I put them in the correct order in the report. (e.g. 1.1. is in `risk`, 1.2 is in `unanswered_risks`, 1.3 is in `risks` again...)

For me, `set_modules_rows` is a logical place to make this distinction: iterate over all the risks in the module, and decide 
1) not applicable? -> skip in the normal report, and save for later 
2) not answered? -> skip in the normal report if we only want risks with answers, and save for later.

I'm trying to be as modular as possible, to support various "flavors" of the report. E.g. the French want to show not answered risks in the normal report, the Italians want to show it in a separate list.

I'm sure there is a better way of coding this, but for the moment, I think it can only be achieved with a substantial re-write of the whole docx module. 
In terms of long-term maintenance, it will probably be worth the effort, especially if you find the current state confusing. I would first start by spending a day or two in writing tests.

Reply:
I have an idea how to make it clearer: I will not define `unanwered_risks` and `not_applicable_risks` as class variables, but as return values of `set_modules_rows`, so that they can then be passed to `add_extra` in `compile`.